cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -305,8 +365,15 @@ int decode_page_headers(cudf::detail::hostdevice_vector<ColumnChunkDesc>& chunks
   DecodePageHeaders(chunks.device_ptr(), chunks.size(), error_code.data(), stream);
 
   if (error_code.value() != 0) {
-    // TODO(ets): if an unsupported encoding was detected, do extra work to figure out which one
-    CUDF_FAIL("Parquet header parsing failed with code(s)" + error_code.str());
+    int32_t mask              = static_cast<int32_t>(decode_error::UNSUPPORTED_ENCODING);
+    bool unsupported_encoding = error_code.value() & mask;
+    if (unsupported_encoding) {
```

Comment:
```suggestion
    if (BitAnd(error_code.value(), decode_error::UNSUPPORTED_ENCODING) != 0) {
```

Reply:
So this will give an SFINAE error likely because `error_code` is not a scoped enumeration. I can add a new overload to deal with this, but I don't have enough context on the codebase to know whether or not this is okay to do. Could you comment? @etseidl

Reply:
So `BitAnd` won't work because it'll return a SFINAE error since the kernel error does not return a scoped enumeration. I can add another overload, but the comments in `BitAnd` clearly say it's only for scoped enumerations so I hesitate to do so. 

I imagine the kernel_error can also be modified, too, but again I don't have enough context on the codebase to know which is appropriate. (Or maybe there's a way to map it into an enumeration that im unaware of?)

Reply:
Ah, probably should have made error_code an unsigned int :( You could just cast `error_code.value()` to a `uint32_t` for now. Might be worth revisiting the `kernel_error` design later.

Reply:
SG! Thank you, will push changes shortly.